remix/app/routes/_app.newsletter.update.$id.test.tsx

```diff
@@ -21,12 +22,12 @@ jest.mock('../config/auth.server', () => ({
 
 jest.mock('../lib/handler/app.handler');
 
-describe('PUT /newsletter/update/:id', () => {
-  beforeEach(() => {
-    jest.clearAllMocks();
-  });
+describe('Newsletter update', () => {
+  describe('loader', () => {
```

Comment:
The previous `describe` block was better. I recommend the following structure:

```javascript
describe('given the user is authenticated', () => {
  describe('given valid newsletter params', () => {
    it('loads the newsletter details', async () => {
      //...
    });
  });

  describe('given valid newsletter updated params', () => {
    it('updates the newsletter', async () => {
      //...
    });
  });

   // nice to have test but not compulsory IMO
   describe('given INVALID newsletter updated params', () => {
    it('returns an error', async () => {
      //...
    });
  });
});

describe('given the user is NOT authenticated', () => {
  it('returns an error', async () => {

  });
})
```

Reply:
Roger that, first I thought separate `action` and `loader` will be good. At the end it cause some duplicate code there.

Improved here: [b357dd3](https://github.com/nimblehq/js-framework-benchmark/pull/130/commits/b357dd39fc2364fe1acf62523fa9f7463e88caed)